go/encoding/basex/encoding.go

```diff
@@ -185,12 +193,12 @@ func (enc *Encoding) decodeBlock(dst []byte, src []byte, baseOffset int) (int, i
 		v := enc.decodeMap[b]
 		si++
 
-		if v.Cmp(invalidByte) == 0 {
+		if v == nil {
```

Comment:
this is the main improvement.  only one branch on the critical path, and not a BigInt operation.